src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -324,7 +326,7 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeFloatArray(vector);
         out.writeInt(k);
         out.writeOptionalNamedWriteable(filter);
-        if (isClusterOnOrAfterMinRequiredVersion("ignore_unmapped")) {
+        if (out.getVersion().onOrAfter(getMinVersionForFeature("ignore_unmapped"))) {
```

Comment:
Does this stream guarantee the min version of the cluster. Just verifying since the previous method has that check

Reply:
Is there a historical reason that we look for the cluster version and not the stream version?
It should behave the same - a feature is on only when both the nodes are upgraded. 

@jmazanec15 / @navneet1v

Reply:
We haven't serialized the stream version when we write KNNQueryBuilder at first. Therefore, we were not able to use stream version. After that, we haven't tried to add stream version because cluster version just worked fine for us.